src/openms/source/SYSTEM/File.cpp

```diff
@@ -89,14 +91,21 @@ namespace OpenMS
     return temp_dir_;
   }
 
+  std::mutex mtx; // mutex for critical section
+
   String File::getExecutablePath()
   {
     // see http://stackoverflow.com/questions/1023306/finding-current-executables-path-without-proc-self-exe/1024937#1024937 for more OS' (if needed)
     static String spath = "";
     static bool path_checked = false;
 
+    const std::lock_guard<std::mutex> lock(mtx);
```

Comment:
I was thinking about a more elegant solution, and I think there is one:
the `static` variable is already protected by a mutex implicitly added by the compiler.
We can make use of this by using an `immediately evaluated lambda` function, like here:
https://stackoverflow.com/questions/15581662/is-it-possible-to-initialize-static-variable-with-lambda

thus, no need for a mutex. and the code becomes easy:
```
String File::getExecutablePath()
{	  
  static String spath = ()[]{ ... code to create path here ... }();
  return spath;
}
```

Reply:
@cbielow  what happens if an exception is thrown? What is the state of the static?

Reply:
its still uninitialized and the next call will try again.